api/server/httputils/errors.go

```diff
@@ -102,3 +109,39 @@ func MakeErrorHandler(err error) http.HandlerFunc {
 		}
 	}
 }
+
+// statusCodeFromGRPCError returns status code according to gRPC error
+func statusCodeFromGRPCError(err error) int {
+	switch grpc.Code(err) {
+	case codes.OK: // code 0
+		return http.StatusOK
+	case codes.InvalidArgument: // code 3
+		return http.StatusBadRequest
+	case codes.DeadlineExceeded: // code 4
+		return http.StatusGatewayTimeout
+	case codes.NotFound: // code 5
+		return http.StatusNotFound
+	case codes.AlreadyExists: // code 6
+		return http.StatusConflict
+	case codes.PermissionDenied: // code 7
+		return http.StatusNotImplemented
+	case codes.Unauthenticated: // code 16
+		return http.StatusUnauthorized
+	case codes.OutOfRange: // code 11
+		return http.StatusRequestedRangeNotSatisfiable
+	case codes.Unimplemented: // code 12
+		return http.StatusNotImplemented
+	case codes.Internal: // code 13
+		return http.StatusInternalServerError
+	case codes.Unavailable: // code 14
+		return http.StatusServiceUnavailable
+	default:
+		// codes.Canceled(1)
+		// codes.Unknown(2)
+		// codes.ResourceExhausted(8)
+		// codes.FailedPrecondition(9)
```

Comment:
well, technically I think 400 is for the HTTP server and shouldn't be used by us, but we do :)